### DISCUSSION TITLE
Is there a way to add a operator to check the length of the values?

### DISCUSSION THREAD

**Discussion by mg-pool** [react-querybuilder/react-querybuilder#475] [Q&A] [answered]
Is there a way to add a length check for the strings, like max_legnth, min_length and exact_length operators?
If I can add the operators, how do I combine them with formatQuery for SQL (MySQL DB) and JSON?

**Comment by jakeboone02** [comment, marked as answer]
You can [manage operators](https://react-querybuilder.js.org/docs/tips/managing-operators) with the `operators` property on each field or the [`getOperators` prop](https://react-querybuilder.js.org/docs/components/querybuilder#getoperators).

To utilize a custom operator in `formatQuery`, use the [`valueProcessor` option](https://react-querybuilder.js.org/docs/utils/export#value-processor).

**Comment by mg-pool** [comment]
```
    const transformedQuery = transformQuery(query, {
        ruleProcessor: r => ({
            ...r,
            field : (r.operator == "max_length" || r.operator == "exact_length" || r.operator == "min_length") ? "LENGTH(" + r.field + ")" : r.field,
            operator : (r.operator == "max_length") ? "<=" : (r.operator == "exact_length") ? "=" : (r.operator == "min_length") ? ">=" : r.operator,
        }),
    });
```

Managed to do it thru transformQuery, thanks for your help

**Reply by jakeboone02** [reply]
Creative!
